pkg/asset/installconfig/platform.go

```diff
@@ -57,61 +61,80 @@ func (a *Platform) Name() string {
 	return "Platform"
 }
 
-func (a *Platform) queryUserForPlatform() string {
-	var platform string
-	survey.AskOne(&survey.Select{
-		Message: "Platform",
-		Options: validPlatforms,
-	}, &platform, nil)
+func (a *Platform) queryUserForPlatform() (string, error) {
```

Comment:
maybe rename this to `queryUserOrGetEnvForPlatform`

Reply:
> maybe rename this to `queryUserOrGetEnvForPlatform`

I'm fine with the old name.  Environment variables are just another channel for the user to answer these queries.

Reply:
So you'd rather double-check each function where it gets the input from instead of having the information readily available in the name?

Reply:
> So you'd rather double-check each function where it gets the input from...

Yeah.  The caller just needs to care that the input is gotten.  The source location is an implementation detail.